File: WEEK 1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int get_positive_int(string prompt); //defining
int main(void)
{
    int i = get_positive_int("Height: ");// seeking height

    for (int height = 0; height < i; height++) //height
    {
        for (int spaces = i - height - 2; spaces >= 0; spaces--)//spaces before
        {
            printf(" ");
        }
        for (int row1 = 0; row1 <= height; row1++)//left pyramid
        {
            printf("#");
        }
        printf("  "); //gap
        for (int row2 = 0; row2 <= height; row2++)// right
        {
            printf("#");
        }
        printf("\n");
    }

}

int get_positive_int(string request)// function defined above
{
    int number;
    do
    {
        number = get_int("%s", request);
    }
    while (number < 1 || number > 8);
    return number;
}

File: WEEK 6/sentimental-mario-more/mario.py
# mario.py

from cs50 import get_int

# Prompt the user for the height of the pyramid until they provide a valid input
while True:
    height = get_int("Height: ")
    if height >= 1 and height <= 8:
        break

# Print the double half-pyramid of the specified height
spaces = height
for i in range(1, height + 1):
    # Print left half of the pyramid
    while spaces > 1:
        ##preceeding spaces
        print(" ",end="")
        spaces = spaces - 1

    for j in range(1, i + 1):
        print("#", end="")
    # reset spaces for next line
    spaces = height - j

    # Print gap between the two halves of the pyramid
    print("  ", end="")

    # Print right half of the pyramid
    for j in range(1, i + 1):
        print("#", end="")

    # Move to the next line
    print()